### ISSUE TITLE
[MDEP-689] Dependency filtering are ignored in go-offline

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-dependency-plugin#1199]
**[Fabrice Bacchella](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=fbacchella)** opened **[MDEP-689](https://issues.apache.org/jira/browse/MDEP-689?redirect=false)** and commented

I'm using a multi-modules project.
I tried as a work around of bug MDEP-688 to tells go-offline to ignore a broken dependency being:

```xml
<dependency>
    <groupId>org.dojotoolkit</groupId>
    <artifactId>dojo</artifactId>
    <type>zip</type>
    <version>1.10.8</version>
    <classifier>distribution</classifier>
</dependency>
```

. But whatever I tried, my filter are ignored:

```java
mvn -B dependency:go-offline -Psonar,versions  -Dmaven.repo.local=/tmp/m2 -DexcludeGroupIds=org.dojotoolkit -DexcludeClassifiers=distribution -DexcludeArtifactIds=org.dojotoolkit:dojo:zip:1.10.8,dojo -DexcludeTypes=zip -Dtype=jar,war,pom -X
...
[DEBUG] Configuring mojo 'org.apache.maven.plugins:maven-dependency-plugin:3.1.2:go-offline' with basic configurator -->
[DEBUG]   (f) appendOutput = false
[DEBUG]   (f) excludeArtifactIds = org.dojotoolkit:dojo:zip:1.10.8,dojo
[DEBUG]   (f) excludeClassifiers = distribution
[DEBUG]   (f) excludeGroupIds = org.dojotoolkit
[DEBUG]   (f) excludeReactor = true
[DEBUG]   (f) excludeTransitive = false
[DEBUG]   (f) excludeTypes = zip
[DEBUG]   (s) markersDirectory = .../jrds-webapp/target/dependency-maven-plugin-markers
[DEBUG]   (f) outputAbsoluteArtifactFilename = false
[DEBUG]   (f) overWriteIfNewer = true
[DEBUG]   (f) overWriteReleases = false
[DEBUG]   (f) overWriteSnapshots = false
[DEBUG]   (s) prependGroupId = false
[DEBUG]   (f) project = MavenProject: fr.jrds:jrds-webapp:2020.1 @ /Users/fa4/Devl/jrds/jrds-webapp/pom.xml
...
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.1.2:go-offline (default-cli) on project jrds-webapp: org.eclipse.aether.resolution.DependencyResolutionException: Could not find artifact org.dojotoolkit:dojo:zip:1.10.8 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```

So I tried everything to prevent the failing artifact to be resolved, nothing succeeded.

---

**Affects:** 3.1.2

**Issue Links:**
- [MDEP-568](https://issues.apache.org/jira/browse/MDEP-568) dependency:go-offline -DexcludeGroupIds=xxxx still try to resolve artifacts in the excluded group xxxx
 (_**"is duplicated by"**_)
- [MDEP-725](https://issues.apache.org/jira/browse/MDEP-725) excludeGroupIds, excludeArtifactIds not working for go-offline goal
 (_**"is duplicated by"**_)

**Remote Links:**
- [GitHub Pull Request #417
](https://github.com/apache/maven-dependency-plugin/pull/417)

2 votes, 8 watchers

**Comment by jira-importer** [comment]
**[Candice Guo](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zzgzy)** commented

Have similar issue. Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.1.2:go-offline on a project. It seems trying to find the artifacts in central repository while I've defined another repository URL for it to find.

**Comment by jira-importer** [comment]
**[Piotr Zygielo](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pzygielo)** commented

> It seems trying to find the artifacts in central repository while I've defined another repository URL for it to find.

[Candice Guo](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zzgzy) - that's because of MDEP-680 I suppose.

**Comment by jira-importer** [comment]
**[Candice Guo](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zzgzy)** commented

[Piotr Zygielo](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pzygielo) The behavior I saw is quite strange. Maven is able to download things based on the url of repository to /.m2/repository folder but it will then throw me error saying: 

`Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.1.2:go-offline (default-cli) on project XYZ: org.eclipse.aether.resolution.DependencyResolutionException: Failure to find com.hortonworks.shc:shc-core:jar:1.1.0.2.6.5.3008-11 in [https://repo.maven.apache.org/maven2] was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced. `

Does it mean maven still trying to download something while it's already being cached?

**Comment by jira-importer** [comment]
**[Candice Guo](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zzgzy)** commented

Hi Piotr, it looks like you are trying to fixed it in maven-dependency-plugin 3.1.2, but the commits have not been merged?
